metomi/rosie/cli_utils.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (C) British Crown (Met Office) & Contributors.
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from subprocess import Popen, PIPE
+import textwrap
+from typing import Optional, List
+
+# NOTE: Box Drawing characters
+# t, b, l, r -> top, bottom, left, right
+# h, v -> horizontal, vertical
+# l, x -> line, cross (i.e. vertex)
+
+# unicode (U_) chars
+U_HL = '─'
+U_VL = '│'
+U_TL = '┌'
+U_TR = '┐'
+U_BL = '└'
+U_BR = '┘'
+U_LX = '├'
+U_RX = '┤'
+U_TX = '┬'
+U_BX = '┴'
+U_XX = '┼'
+
+# ASCII (A_) chars
+A_HL = '-'
+A_VL = '|'
+A_TL = '+'
+A_TR = '+'
+A_BL = '+'
+A_BR = '+'
+A_LX = '+'
+A_RX = '+'
+A_TX = '+'
+A_BX = '+'
+A_XX = '+'
+
+
+def get_terminal_cols(default=80):
+    """Return the width of the terminal in columns.
+
+    If the check fails, this will fallback to the default value.
+    """
+    proc = Popen(['stty', 'size'], stdout=PIPE, stderr=PIPE, text=True)
+    try:
+        return int(proc.communicate()[0].split()[1]) or default
+    except Exception:
+        return default
+
+
+def table(
+    rows: List[List[str]],
+    header: Optional[List[str]] = None,
+    max_width: Optional[int] = None,
+    unicode: bool = True,
+) -> str:
+    """Format text into a table.
+
+    Args:
+        rows: 2D table composed of lists.
+        header: Optional 1D list to use as the table header.
+        max_width: Maximum permitted width for the whole table including
+            borders.
+        unicode: Use unicode characters if True, else fallback to ascii.
+
+    Examples:
+        >>> print(table(['a', 'b', 'c']))
```

Comment:
Should this work too? (It doesn't)
```suggestion
        >>> print(table(['aleph', 'bandicoot', 'coot']))
```

Reply:
Removed the 1D table example.